compiler/rustc_trait_selection/src/traits/select/mod.rs

```diff
@@ -591,7 +591,9 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
                     }
                 }
             }
-            ty::PredicateAtom::TypeWellFormedFromEnv(..) => unimplemented!(),
+            ty::PredicateAtom::TypeWellFormedFromEnv(..) => {
+                bug!("TypeWellFormedFromEnv is only used for chalk")
```

Comment:
@jackh726 as this didn't get hit in the test suite, I guessed that this would be the right solution. Could you check?

Reply:
This is good.